### PR TITLE
Adjust process_create and reuse pagetable_getmap in run()

### PR DESCRIPTION
**Pull request by zachwaterson**
It was discovered that we needed to pass in the size of the code to process_create() in order for it to allocate enough pages for the process data to be stored at the entry point.

In the same vein, we need to pagetable_getmap() for each chunk of process we move over, since we might be going into a new page.

### CODE REVIEW

File: src/syscall_handler_process.c [hunk 1]
```diff
@@ -10,7 +10,7 @@ See the file LICENSE for details.
 #include "memorylayout.h" // PROCESS_ENTRY_POINT
 #include "permissions_capabilities.h"
 
-#define PROCESS_COPY_CHUNK 3500
+#define PROCESS_COPY_CHUNK 2048 // Half a page
```
Comment (kkoser) [inline src/syscall_handler_process.c]: can we use PAGE_SIZE / 2 here?

Comment (zachwaterson) [inline src/syscall_handler_process.c]: Good thought; I didn't realize PAGE_SIZE was in the namespace of the handler. Updating now

**Comment by alexmwu** [comment]
why page_size/2 just out of curiosity?

**Comment by kkoser** [comment]
It needs to be a number that will go evenly into PAGE_SIZE so that we don't have to deal with stopping copy chunks that would cross pages.

**Comment by alexmwu** [comment]
ah and < kmalloc size got it.

**Comment by alexmwu** [comment]
LGTM

**Pull request merged**